src/pyfaf/actions/repolist.py

```diff
@@ -28,12 +28,26 @@ class RepoList(Action):
     def run(self, cmdline, db):
         if cmdline.detailed:
             data = []
-            header = ["Name", "Type", "URL", "Nice name"]
+            header = ["Name", "Type", "URL", "Nice name", "GPG check", "OS", "OS release", "Architecture"]
             for repo in db.session.query(Repo):
                 data.append((repo.name, repo.type, repo.url_list[0].url,
-                             repo.nice_name or ""))
-                for url in repo.url_list[1:]:
-                    data.append(("", "", url.url, ""))
+                             repo.nice_name or "", str(not repo.nogpgcheck),
+                             repo.opsys_list[0] if repo.opsys_list else "",
+                             repo.opsysrelease_list[0] if repo.opsysrelease_list else "",
+                             repo.arch_list[0] if repo.arch_list else ""))
+                for i in range(1, len(max([repo.url_list,
+                                           repo.opsys_list,
+                                           repo.opsysrelease_list,
+                                           repo.arch_list],
+                                          key=len))):
+                    data.append(("",
+                                 "",
+                                 repo.url_list[i].url if len(repo.url_list) > i else "",
+                                 "",
+                                 "",
+                                 repo.opsys_list[i] if len(repo.opsys_list) > i else "",
```

Comment:
I think you are not able to assign repo to OpSys only to OpSysRelease, so you can remove the `OS` column.

I've tried with 
```
$ faf repoassign hello Fedora
[2020-02-05 15:01:34] ERROR:faf.RepoAssign: Assigning operating system without release to non - parametrized repo. Assign operating system with release.
```

```
bash-5.0$ faf repolist --detailed
Name      Type  URL                                                                                                                                          Nice name  GPG check  OS  OS release  Architecture  
f31       dnf   file:///tmp                                                                                                                                             False          Fedora 32   x86_64        
                asdf                                                                                                                                                                                             
f32       dnf   file:///tmp                                                                                                                                             False                                    
                ftp://super-fricking-loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong-addresssssssssssssssssssssssssssssssssssss-ooooo                                                      
f32_??    dnf   file:///tmp                                                                                                                                             True                                     
f32_**??  dnf   file:///tmp                                                                                                                                             True                                     
hello     dnf   file:///tmp                                                                                                                                             True           Fedora 31   x86_64        
                asdf                                                                                                                                                                   Fedora 32   i386          
                                                                                                                                                                                       Fedora 30
```

Reply:
But the following works:  
  
    $ faf repoadd hello dnf file:///repo/releases/$  
    $ faf repoassign hello Fedora  
    $ faf repolist --detailed  
    Name          Type         URL                      Nice name  GPG check  OS      OS release  Architecture  
    hello         dnf          file:///repo/releases/$             True       Fedora  
  
Can't this ever happen in practice?

Reply:
Oh, you are right.